### PR TITLE
Feat: make Haproxy captures ECS compliant

### PR DESCRIPTION
**Pull request by kares**
The naming is, yet again, inspired by Beats' [haproxy module](https://github.com/elastic/beats/blob/v7.9.2/filebeat/module/haproxy/log/ingest/pipeline.yml). 
As usual Beats can do more using the whole pipeline but we're pretty close, notable differences:

`Nov  4 08:28:42 debian10 haproxy[3666]: 127.0.0.1:34278 [04/Nov/2020:08:28:42.475] front bnodes/node1 0/0/0/0/1 200 10956 - - ---- 1/1/0/1/0 0/0 "GET / HTTP/1.1"`

- Beats matches the *debian10* part as `haproxy.source` while LS goes with `IPORHOST:[host][hostname]`
- Beats does not capture `[haproxy][total_time_ms]` -> this fields ends up as `event.duration` (after being * 1000),
  represents the `%Tt` field from haproxy's log line [format](https://www.haproxy.com/wp-content/uploads/2017/07/aloha_load_balancer_memo_log.pdf)
  * LS also does not type-cast this field on purpose as it might be `+` prefixed in which case it indicates an "inaccuracy"
- Beats does not parse out the [raw request line](https://github.com/elastic/beats/blob/v7.9.2/filebeat/module/haproxy/log/test/haproxy.log-expected.json#L27) (*"GET / HTTP/1.1"* part) while LS will do into:
  `http.request.method`, `http.version` and `url.*` fields

### CODE REVIEW

**Comment by kares** [comment]
@ebeahan @webmat if you guys find time to have a look, as outlined in the PR's description it's close to what Beats does - 
with the few differences noted. the `HAPROXYTCP` is a stripped down :scissors: version of `HAPROXYHTTP` so it's mostly about matching the http log, sample match: https://github.com/logstash-plugins/logstash-patterns-core/blob/f4e7242e47a9b11c4bb482078811c6603ebf8250/spec/patterns/haproxy_spec.rb#L9-L35

**Comment by webmat** [comment]
I have a *complaint* about the title of this PR 😂

**Review by webmat — APPROVED** [review]
I'm good with all 3 points you've laid out in the PR body 👍 

I've made a single comment, not sure if this is applicable for HAProxy, so feel free to adjust or disregard :-)

LGTM otherwise

File: patterns/ecs-v1/haproxy [hunk 1]
```diff
@@ -30,10 +25,15 @@ HAPROXYCAPTUREDRESPONSEHEADERS %{DATA:captured_response_headers}
...
+HAPROXYTCP (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp}) %{IPORHOST:[host][hostname]} %{SYSLOGPROG}: %{IP:[source][address]}:%{INT:[source][port]:int} \[%{HAPROXYDATE:[haproxy][request_date]}\] %{NOTSPACE:[haproxy][frontend_name]} %{NOTSPACE:[haproxy][backend_name]}/(?:<NOSRV>|%{NOTSPACE:[haproxy][server_name]}) (?:-1|%{INT:[haproxy][total_waiting_time_ms]:int})/(?:-1|%{INT:[haproxy][connection_wait_time_ms]:int})/%{NOTSPACE:[haproxy][total_time_ms]} %{NOTSPACE:[haproxy][bytes_read]:int} %{NOTSPACE:[haproxy][termination_state]} %{INT:[haproxy][connections][active]:int}/%{INT:[haproxy][connections][frontend]:int}/%{INT:[haproxy][connections][backend]:int}/%{INT:[haproxy][connections][server]:int}/%{NOTSPACE:[haproxy][connections][retries]:int} %{INT:[haproxy][server_queue]:int}/%{INT:[haproxy][backend_queue]:int}
```
[the remark this answers is not in the document]
Comment (kares) [reply patterns/ecs-v1/haproxy:38]: thanks - from the docs it stands for "total number of bytes transmitted to the client when the log is emitted" 
sounds good enough for me to match as `source.bytes`, will adjust.

File: patterns/ecs-v1/haproxy [hunk 2]
```diff
@@ -30,10 +25,15 @@ HAPROXYCAPTUREDRESPONSEHEADERS %{DATA:captured_response_headers}
 # HAPROXYCAPTUREDREQUESTHEADERS %{DATA:request_header_host}\|%{DATA:request_header_x_forwarded_for}\|%{DATA:request_header_accept_language}\|%{DATA:request_header_referer}\|%{DATA:request_header_user_agent}
```
Comment (yaauie) [inline patterns/ecs-v1/haproxy:25]: nit: even though these are commented out, we should update the examples to use ECS-ish names.

**Review by yaauie — APPROVED** [review]
One nit about updating the a commented-out example, otherwise LGTM.

:tada: for the specs, which tell the story and show the structure well.

**Pull request merged**